jormungandr/src/blockchain/tip.rs

```diff
@@ -202,52 +202,69 @@ impl TipUpdater {
         Ok(())
     }
 
-    /// this function will re-process the tip against the different branches
+    /// this function will re-process the tip against the different branches.
     /// this is because a branch may have become more interesting with time
     /// moving forward and branches may have been dismissed
-    async fn reprocess_tip(&mut self) -> Result<(), Error> {
-        let branches: Vec<Arc<Ref>> = self.blockchain.branches().branches().await;
-        let tip_as_ref = self.tip.get_ref().await;
-
-        let others = branches
-            .iter()
-            .filter(|r| !Arc::ptr_eq(r, &tip_as_ref))
-            .collect::<Vec<_>>();
+    #[instrument(level = "debug", skip_all, fields(current_tip = %tip.header().description()))]
+    async fn reprocess_tip(
+        blockchain: Blockchain,
+        tip: Arc<Ref>,
+        mut mbox: MessageBox<Arc<Ref>>,
+    ) -> Result<(), Error> {
+        use std::cmp::Ordering;
+        let branches = blockchain.branches().await?;
+        let storage = blockchain.storage();
+
+        let best_branch = branches.into_iter().map(Branch::into_ref).max_by(|a, b| {
```

Comment:
This may add some bias to hashes with lower leading bits, I think? At least I think that's the order of iteration.

Not sure if it matters, though. And it may be more a thing with the `compare_against` function.

Reply:
> This may add some bias to hashes with lower leading bits, I think

Why should it matter? the max function (in this case tip selection) does not really depend on the order (apart from equal score chains, but in this case I guess it's arbitrary the one you choose)

Reply:
Well... It may not matter. I meant for equal score chains. The storage is a btree, so I think it's sorted, and this will pick the first in that order. 

It doesn't matter for fund7 though, so I'm not proposing to change it. What I was wondering is whether it could incentivize pools to produce blocks with small hashes.

Reply:
In a standard flow, tips would go first through the normal processing function, where time is the biggest advantage, so I guess the additional work to produce blocks with small hashes would not be beneficial. 
Good call out though, didn't think of that